File: Day75/Part4/trap_activation.py
def solve():
    try:
        line = input().split()
        if not line:
            return
        
        if len(line) == 1:
            n = int(line[0])
            line2 = input().split()
            t = [int(x) for x in line2]
        else:
            t = [int(x) for x in line]
            
        if not t:
            print(0)
            return
            
        max_active = 0
        current = 0
        
        for trap in t:
            if trap == 1:
                current += 1
                max_active = max(max_active, current)
            else:
                current = 0
                
        print(max_active)
            
    except EOFError:
        pass

if __name__ == "__main__":
    solve()

File: Day75/Part4/trap_activation.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    if (cin >> n) {
        if (n == 0) {
            cout << 0 << endl;
            return 0;
        }
        vector<int> t(n);
        for (int i = 0; i < n; ++i) cin >> t[i];

        // Find longest contiguous segment of 1s
        int max_active = 0;
        int current = 0;
        
        for (int i = 0; i < n; ++i) {
            if (t[i] == 1) {
                current++;
                max_active = max(max_active, current);
            } else {
                current = 0;
            }
        }

        cout << max_active << endl;
    }
    return 0;
}
